src/handlers/psp.ts

```diff
@@ -0,0 +1,436 @@
+      [
+        subscription.chainId,
+        subscription.airnodeAddress,
+        subscription.templateId,
+        subscription.parameters,
+        subscription.conditions,
+        subscription.relayer,
+        subscription.sponsor,
+        subscription.requester,
+        subscription.fulfillFunctionId,
+      ]
+    );
+    if (subscriptionId !== expectedSubscriptionId) {
+      node.logger.warn(
+        `SubscriptionId ${subscriptionId} does not match expected ${expectedSubscriptionId}`,
+        baseLogOptions
+      );
+      return;
+    }
+
+    enabledSubscriptions.push({
+      ...subscription,
+      id: subscriptionId,
+    });
+  });
+
+  const groupedSubscriptions: GroupedSubscriptions[] = [];
+  if (isEmpty(enabledSubscriptions)) {
+    node.logger.info('No proto-psp subscriptions to process', baseLogOptions);
+  } else {
+    const enabledSubscriptionsByTemplateId = groupBy(enabledSubscriptions, 'templateId');
+    Object.keys(enabledSubscriptionsByTemplateId).forEach((templateId) => {
+      // Get template details
+      const template = config.templates[templateId];
+      if (isNil(template)) {
+        node.logger.warn(`TemplateId ${templateId} not found in templates`, baseLogOptions);
+        return;
+      }
+      // Verify templateId
+      const expectedTemplateId = ethers.utils.solidityKeccak256(
+        ['bytes32', 'bytes'],
+        [template.endpointId, template.templateParameters]
+      );
+      if (expectedTemplateId !== templateId) {
+        node.logger.warn(`TemplateId ${templateId} does not match expected ${expectedTemplateId}`, baseLogOptions);
+        return;
+      }
+
+      // Get endpoint details
+      const endpoint = config.endpoints[template.endpointId];
+      if (isNil(endpoint)) {
+        node.logger.warn(`EndpointId ${template.endpointId} not found in endpoints`, baseLogOptions);
+        return;
+      }
+      // Verify endpointId
+      const expectedEndpointId = ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(['string', 'string'], [endpoint.oisTitle, endpoint.endpointName])
+      );
+      if (expectedEndpointId !== template.endpointId) {
+        node.logger.warn(
+          `EndpointId ${template.endpointId} does not match expected ${expectedEndpointId}`,
+          baseLogOptions
+        );
+        return;
+      }
+
+      groupedSubscriptions.push({
+        subscriptions: enabledSubscriptionsByTemplateId[templateId],
+        template: { ...template, id: templateId },
+        endpoint: { ...endpoint, id: template.endpointId },
+      });
+    });
+  }
+
+  return {
+    config,
+    baseLogOptions,
+    groupedSubscriptions,
+    apiValuesBySubscriptionId: {},
+    providerStates: [],
+  };
+};
+
+const executeApiCalls = async (state: State): Promise<State> => {
+  const { config, baseLogOptions, groupedSubscriptions } = state;
+
+  // TODO: promise.all? 🤔
```

Comment:
;) It'll work well here to do this as `Promise.all` as it'll make the apiCalls async, which are I/O heavy (would greatly benefit from this behaviour).

I have a suspicion that it is unnecessary to split out callApi into a separate Lambda. Only crypto operations are compute-heavy enough to warrant being split out.

```
  const promisedApiValuesBySubscriptionId = groupedSubscriptions
    .map(async ({ subscriptions, template, endpoint }) => {
    const templateLogOptions: node.LogOptions = {
      ...baseLogOptions,
      additional: {
        templateId: template.id,
      },
    };
```

Reply:
With Michal we were wondering if we could reuse the callApi() function from the airnode-node package.  Maybe that would be enough and we don't even need to start a new function lambda for making api calls if it's not a bottleneck.

Reply:
That sounds pretty cool :)